### ISSUE TITLE
Application life cycle doc should include more real-life details

### ISSUE THREAD

**Issue by siamaksade** [openshift/openshift-docs#3289]
The current app lifecycle doc `dev_guide/application_lifecycle.adoc` presents simple scenarios leaving out the practical details of dealing with OpenShift during the development phase. It needs to include more details on the process. 

**Developing on OpenShift** 
This example should be broken down into a scenario when there is no app skeleton and git repo (this one is covered) and when there already is a git repo with the skeleton (not covered but is the majority of use-cases). Also the develop and verify step should be presented as iterative and not serial:

Developing a New App on OpenShift
```
                                                                      +----------------------+
                                                                      |                      |
                                                                      |                      |
                                                                      v                      |
Plan -> Create Git Repo -> Access OpenShift -> Create Template -> Develop -> Configure -> Verify
```

Developing a Existing App on OpenShift
```
                                          +----------------------+
                                          |                      |
                                          |                      |
                                          v                      +
Clone Git Repo -> Access OpenShift -> Develop -> Configure -> Verify
```

We also should include the following:
* Git repos with credentials should be explained or linked to the docs explaining them.
* In the description of _Configure_, we need to mention builds, deployments, configmap, env vars, secrets, etc with links to their docs. 
* The _Create Template_ should include links to the template docs and also give recommendation for developers how to create a template. Most apps will have to create a template to deploy them with proper configs, etc. 
* How the iterations done should be elaborated more. How is it done with JBDS and using JBoss Tools? How to use `rsync -w`? There should be instructions or link to instructions of how exactly use should do these stuff.

**Comment by bfallonf** [comment]
Thanks @siamaksade 

I can see you've commented on #3275 as well, but I'll include this issue there. It might be able to be tackled with that PR as well.

**Comment by vikram-redhat** [comment]
This was fixed via #3275. Closing.
